packages/daimo-userop/src/daimoOpBuilder.ts

```diff
@@ -29,59 +36,64 @@ function getSigningMiddleware(
   };
 }
 
+// Metadata for a userop: nonce and paymaster constant.
+export type DaimoOpMetadata = {
+  nonce: DaimoNonce;
+  chainGasConstants: ChainGasConstants;
+};
+
 /** Creates userops from a Daimo account.  */
 export class DaimoOpBuilder extends UserOperationBuilder {
   /** Connection to the chain */
   private provider: BundlerJsonRpcProvider;
 
-  private gasMiddleware: UserOperationMiddlewareFn;
-
   /** Daimo account address */
-  address: `0x${string}`;
+  address: `0x${string}` = "0x";
 
-  private constructor(_paymasterMiddleware?: UserOperationMiddlewareFn) {
+  private constructor(rpcUrl: string) {
     super();
-    this.provider = new BundlerJsonRpcProvider(config.rpcUrl).setBundlerRpc(
+    this.provider = new BundlerJsonRpcProvider(rpcUrl).setBundlerRpc(
       config.bundlerRpcUrl
     );
-    this.address = "0x";
-    this.gasMiddleware =
-      _paymasterMiddleware ||
-      Presets.Middleware.estimateUserOperationGas(this.provider);
   }
 
   /** Client is used for simulation. Paymaster pays for userops. */
   public static async init(
     deployedAddress: Address,
-    paymasterMiddleware: UserOperationMiddlewareFn | undefined,
-    signUserOperation: SigningCallback
+    signUserOperation: SigningCallback,
+    rpcUrl: string
   ): Promise<DaimoOpBuilder> {
-    const instance = new DaimoOpBuilder(paymasterMiddleware);
+    const instance = new DaimoOpBuilder(rpcUrl);
     instance.address = deployedAddress;
 
     console.log(`[OP]: init address ${instance.address}`);
     const base = instance
       .useDefaults({
         sender: instance.address,
         signature: dummySignature,
-        verificationGasLimit: 2000000n,
-        callGasLimit: 1000000n,
-      })
-      .useMiddleware(Presets.Middleware.getGasPrice(instance.provider))
-      .useMiddleware(instance.gasMiddleware)
-      .useMiddleware(async (ctx) => {
-        ctx.op.verificationGasLimit = 2000000n;
-
-        // Workaround: Pimlico gas price estimator seems to be too low
-        // ctx.op.callGasLimit = Math.floor(Number(ctx.op.callGasLimit) * 10);
+        verificationGasLimit: DEFAULT_USEROP_VERIFICATION_GAS_LIMIT,
+        callGasLimit: DEFAULT_USEROP_CALL_GAS_LIMIT,
+        preVerificationGas: DEFAULT_USEROP_PREVERIFICATION_GAS_LIMIT,
```

Comment:
looks great

and verificationGasLimit doesn't need to be 2m anymore

Reply:
good point, weirdly it seems like any number < 1.5m actually fails the paymaster with "AA33 reverted: PP-ERC20 : token amount too high". Will ask Kristoff.